src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1043,28 +1082,43 @@ MPS_Tensor MPS::state_vec_as_MPS(uint_t first_index, uint_t last_index) const
 	}
 	// now temp is a tensor of 2^n matrices of size 1X1
 	temp.mul_Gamma_by_right_Lambda(right_lambda);
-
 	return temp;
 }
 
-void MPS::full_state_vector(cvector_t& statevector) const
-{
-  MPS_Tensor mps_vec = state_vec_as_MPS(0, num_qubits_-1);
-  uint_t length = 1ULL << num_qubits_;   // length = pow(2, num_qubits_)
-  statevector.resize(length);
+void MPS::full_state_vector(cvector_t& statevector){
+  reg_t qubits(num_qubits_);
+  std::iota( std::begin(qubits), std::end(qubits), 0);
+  reg_t internal_qubits = get_internal_qubits(qubits);
+  full_state_vector_internal(statevector, internal_qubits);
+}
+
+void MPS::full_state_vector_internal(cvector_t& statevector,
+				     const reg_t &qubits) {
+  // mps_vec contains the state vector with the qubits in ascending order
+  MPS_Tensor mps_vec = state_vec_as_MPS(qubits);
 
+  uint_t num_qubits = qubits.size();
+  uint_t length = 1ULL << num_qubits;   // length = pow(2, num_qubits)
+  statevector.resize(length);
+  // statevector is constructed in ascending order
 #pragma omp parallel for if (num_qubits_ > omp_threshold_ && omp_threads_ > 1) num_threads(omp_threads_)
   for (int_t i = 0; i < static_cast<int_t>(length); i++) {
-    statevector[i] = mps_vec.get_data(reverse_bits(i, num_qubits_))(0,0);
+    statevector[i] = mps_vec.get_data(i)(0,0);
   }
+  cvector_t temp_statevector(length);
+  //temp_statevector will contain the statevector in the ordering define in "qubits"
+  reorder_all_qubits(statevector, qubits, temp_statevector);
+  // reverse to be consistent with qasm ordering
+  statevector = reverse_all_bits(temp_statevector, num_qubits);
+}
 
-#ifdef DEBUG
-  std::cout << *this;
-#endif
+void MPS::get_probabilities_vector(rvector_t& probvector, const reg_t &qubits) {
```

Comment:
If we can make this one `const` would be great too.

Reply:
Returned the const.